src/cmds/blockdev.c

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file
+ * @brief command line utility, to show information w.r.t block devices
+ *
+ * @date 20.02.23
+ * @author Adityar Rayala
+ */
+
+#include<stdio.h>
+#include<string.h>
+#include<assert.h>
+#include <drivers/block_dev.h>
+#include <getopt.h>
+
+static void print_usage(void) {
+	printf("Usage:\nblockdev command devices\n");
+	printf("blockdev --report [devices]\n");
+	printf("Options:\n");
+	printf(" %-15s print report for specified (or all) devices.\n", "--report");
+	printf("Available Commands:\n");
+	printf(" %-15s get size in 512-byte sectors.\n", "--getbsz");
+}
+
+static inline struct block_dev* chk_dev_avail(char* dev_name) {
+	struct block_dev *bdev;
+
+	bdev = block_dev_find(dev_name);
+	if (!bdev) {
+		printf("Block device \"%s\" not found.\n", dev_name);
+		return 0;
+	}
+
+	return bdev;
+}
+
+int main(int argc, char **argv) {
+	struct block_dev **bdevs, *bdev;
+	struct option longopts[] = {
+		{ "getbsz", required_argument, 0, 1 },
+		{ "report", optional_argument, 0, 2 },
+	};
+	int opt;
+
+	if (argc < 2) {
+		print_usage();
+		return 0;
+	}
+
+	if (-1 != (opt = getopt_long(argc, argv, "", longopts, NULL))) {
+		switch (opt) {
+			case 1:
+				if (argc < 3){
+					printf("blockdev: No device specified\n");
+				}
+				else{
+					bdev = chk_dev_avail(argv[2]);
+					if (bdev) {
+						printf("%d\n", bdev->block_size);
+					}
+				}
+				break;
+			case 2:
+				printf("------------------------------------------------------------\n");
+				printf("| %-15s | %-15s | %-20s |\n", "Device Name", "BSZ", "Size");
+				printf("------------------------------------------------------------\n");
+				bdevs = get_bdev_tab();
+				assert(bdevs);
```

Comment:
Please, don't use assert in command. Use 'if' instead

Reply:
Ok, I will change and update it.